system_metrics_collector/src/system_metrics_collector/collector.cpp

```diff
@@ -12,14 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "collector.hpp"
+
+#include <mutex>
+#include <sstream>
+#include <string>
 
 #include "../../src/moving_average_statistics/moving_average.hpp"
 #include "../../src/moving_average_statistics/types.hpp"
 
-#include "collector.hpp"
+bool Collector::start()
+{
+  std::unique_lock<std::mutex> ulock(mutex);
+  if (started_) {
+    return false;
+  }
+  started_ = true;
+  return setupStart();
+}
+
+bool Collector::stop()
+{
+  std::unique_lock<std::mutex> ulock(mutex);
+  if (!started_) {
+    return false;
+  }
+  started_ = false;
 
+  const bool ret = setupStop();
+  clearCurrentMeasurements();
```

Comment:
does that need to be a single lock?

Reply:
Do you mean that clearCurrentMeasurements() doesn't need to be locked in this scope, because it's already protected? That makes sense to me, while keeping setupStop in the locked scope as a private method.